Userland/Libraries/LibCompress/Lzma.h

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023, Tim Schumacher <timschumi@gmx.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/CircularBuffer.h>
+#include <AK/FixedArray.h>
+#include <AK/NonnullOwnPtr.h>
+#include <AK/Stream.h>
+
+namespace Compress {
+
+// This implementation is mostly based on the LZMA specification contained in the 7-Zip SDK, which has been placed in the public domain.
+// LZMA Specification Draft (2015): https://www.7-zip.org/a/lzma-specification.7z
+
+struct LzmaDecompressorOptions {
+    u8 literal_context_bits;
+    u8 literal_position_bits;
+    u8 position_bits;
+    u32 dictionary_size;
+    Optional<u64> uncompressed_size;
+};
+
+class LzmaDecompressor : public Stream {
+public:
+    static ErrorOr<NonnullOwnPtr<LzmaDecompressor>> create(NonnullOwnPtr<Stream>, LzmaDecompressorOptions const&);
```

Comment:
Should this not take `Handle<Stream>`?

Reply:
It's called `MaybeOwned` now, but yes, it should (at the very least for LZMA2). :^)